### PR TITLE
Webgl fixes

### LINKED ISSUE #20372: Investigate if GetTexParameter should fail if 0 is returned.
Currently `GetTexParameter` fails if 0 is returned, but none of the specs specify that.
Specs:
[WebGL](https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.8)
[GLES](https://www.khronos.org/registry/OpenGL/specs/es/2.0/es_full_spec_2.0.pdf#nameddest=section-6.1.3)
https://github.com/servo/servo/blob/730bd5ec807fec0ed65bbf85afebe3bbca64b32c/components/canvas/webgl_thread.rs#L1075-L1077
https://github.com/servo/servo/blob/730bd5ec807fec0ed65bbf85afebe3bbca64b32c/components/script/dom/webglrenderingcontext.rs#L1362-L1364

### LINKED ISSUE #20553: Implement gl.getParameter(gl.UNPACK_FLIP_Y_WEBGL)
https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.3
https://www.khronos.org/registry/OpenGL-Refpages/es2.0/xhtml/glGet.xml

### LINKED ISSUE #20554: Implement gl.getParameter(gl.UNPACK_PREMULTIPLY_ALPHA_WEBGL)
https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.3
https://www.khronos.org/registry/OpenGL-Refpages/es2.0/xhtml/glGet.xml

### PR DESCRIPTION
**Pull request by anholt**
Add gl.getParameter() support for webgl unpack settings, and remove some unreachable and incorrect error-generation code 

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #20372 
- [x] These changes fix #20553 
- [x] These changes fix #20554 

- [x] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/20754)

### CODE REVIEW

File: components/script/dom/webglrenderingcontext.rs [hunk 1]
```diff
@@ -1326,6 +1326,17 @@ impl WebGLRenderingContextMethods for WebGLRenderingContext {
                 "WebGL GLSL ES 1.0".to_jsval(cx, rval.handle_mut());
                 return rval.get();
             }
+
```
Comment (emilio) [inline components/script/dom/webglrenderingcontext.rs]: nit: the rest of the branches don't have newlines around, so I'd remove them.

**Review by emilio — APPROVED** [review]
Thank you!

**Comment by anholt** [comment]
r+

(am I doing this right?)

**Comment by jdm** [comment]
@bors-servo retry
#20756

**Comment by emilio** [comment]
@bors-servo retry

 * The log appears empty?

**Comment by nox** [comment]
@bors-servo retry 

@jdm Your tool seems unhappy:

```
HTTPError: HTTP Error 500: INTERNAL SERVER ERROR
```

**Comment by nox** [comment]
That's #20756 again. Will retry once the other builds finish.

**Comment by nox** [comment]
@bors-servo retry

That's #20756 again.

**Comment by nox** [comment]
@bors-servo retry

I've been told that the intermittent tracker is fixed.

**Comment by jdm** [comment]
@bors-servo retry
* #20756

**Pull request merged**